agent/internal/docker.go

```diff
@@ -127,6 +127,17 @@ func (d *dockerActor) pullImage(ctx *actor.Context, msg pullImage) {
 			sendErr(ctx, errors.Wrap(err, "error encoding registry credentials"))
 			return
 		}
+	} else if store, ok := newCredentialStore(ctx, reference.Domain(ref)); ok {
```

Comment:
Do we want the store to be a global object? This will initialize a store (which reads the docker config file) for every container.